### PR TITLE
SoC final report blog

### PR DESCRIPTION
**Pull request by kxz18**
Fixes # .

Make sure these boxes are checked before creating your pull request -- thank you!

- [x] Add Blog Header, including title, author, date.  
    For example:  

    ``` md
    ---
    title: "Your Great Blog Title"
    author: your_github_id
    image: /assets/2020/08-your-blog-slug/teaser.jpg
    ---

    YOUR_FIRST_SENTENCE_SHOULD_BE_A_SUMMARY_FOR_YOUR_POST
    ```

- [x] Keep all filenames & url as lowercase, and use `-` to connect words instead of space. e.g. `2017-10-06-wechat-pc-impactor` instead of `2017-10-06-WeChat PC Impactor`,
- [x] Embed the photo & video before publishing, save all external file to a folder named by `${MONTH}-${YOUR_BLOG_SLUG}/` under `/assets/2020` directory.
- [x] Add teaser image for the blog
- [x] Add your self as an author to the author data file at <https://github.com/wechaty/wechaty.js.org/blob/master/_developers/YOUR_GITHUB_USERNAME.md> with your GitHub username. And please also remember to reference yourself with the Github username in your blog post, so that we can have nice author information besides your blog after we published it!
  For excample:
  ```md
  ---
  name: Jiarui LI (李佳芮)
  site: https://lijiarui.github.io
  bio: 句子互动创始人 & CEO，微软人工智能最具价值专家 (AI MVP)
  avatar: /assets/developers/lijiarui/avatar.png
  email: rui@juzi.bot
  twitter: https://weibo.com/u/2175505900
  ---
  ```
  Here is the example file:
  - lijiarui.md: https://github.com/wechaty/wechaty.github.io/blob/master/_developers/lijiarui.md
  - It will shown like this on wechaty blog: https://wechaty.github.io/developers/lijiarui/

To read more about the contribute guideline, visit: <https://github.com/wechaty/wechaty.js.org#contribute-guideline>

### CODE REVIEW

**Review by wj-Mcat — CHANGES_REQUESTED** [review]
You have a great & fast blog. Waiting for your comment from mentor.

File: jekyll/_posts/2020-09-26-chassist-final.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+
+ title: "暑期2020 [基于python-wechaty的群聊助手机器人] 结项成果展示"
+ date: 2020-08-14 09:00 +0800
+ author: kxz18
+ categories: project
+ image: /assets/2020/groupchat-assistant-bot-final/logo.png
+ tags:
+   - python
+   - soc2020
```
Comment (lijiarui) [inline jekyll/_posts/2020-09-26-chassist-final.md:11]: Please add a `soc` tag here.

File: jekyll/_posts/2020-09-26-chassist-final.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  - 对交互方面进行统一优化，提高解析用户命令时的容错性，尝试用深度学习方法对问答系统的匹配准确性进行提升（在基础版本中使用 cosine 相似度对问题和答案进行匹配）、在用户命令解析失败但和某些模式很相近时进行确认等。同时该过程中会不断模拟用户场景发现机器人在交互上不够人性化的地方并进行修正。
+  - 扩展功能
+
+## 二、项目总结
+
+### 2.1 项目成果
+
+目前三阶段的任务已经完成，需求的功能已经实现，相应的单元测试也已经实现，并且配置了 github 的 CI 流程，用于代码提交的检测。同时为项目的 github 项目提供了 readme 进行使用说明，也设计了项目 Logo 和项目名称，项目名称为CHAssisT，寓意为 chat + assistant，即聊天助手。Logo 设计如下图：
+
+<img src="/home/kxz/Documents/Study/MeMyself/Summer2020/blogs/final/images/logo.png" alt="logo" style="zoom: 33%;" />
```
Comment (lijiarui) [inline jekyll/_posts/2020-09-26-chassist-final.md]: Where is this image?

File: jekyll/_posts/2020-09-26-chassist-final.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+### 2.2 项目心得
+
+进行本项目的整个过程可以看成是简单的软件工程实践，遇到的问题及解决的方式也与软件工程中的常见问题及解决类似。
+
+一开始是进行需求的确认。因为原始的需求只是一些比较朦胧的想法，很多细节都没有落实，所以开始项目的第一步就是和导师确认各种需求的细节。在确认需求的时候，首先是进行了一次讨论，对原始的想法有了大致的了解，列了一个大致的需求文档。但是光有需求文档对于具体实现是不够的，因此又采取了用户故事的模式对需求细节进行确定。用户故事即模拟一个用户，逐一模拟各个功能的使用，将使用过程及过程中遇到的问题记录下来，用于在具体实现的时候进行参照。
+
+之后是进行软件架构，因为需求比较多，显然如果直接动手写的话可能会导致无从下手、重复代码多、可维护性差、拓展性差等问题，所以在进行实际的代码书写之前先根据功能进行了模块的划分，用 uml 图的方式进行了结构的确立。这样的方式可以把庞大的需求分解成一个个子任务，同时可以将普适的结构抽象成接口，方便实现的同时也可以享有较规整的代码结构带来的益处。这一步需要花费的精力通常会比实际写代码花费的精力还多，因为好的架构可以事半功倍，反之亦然。当然，在动工前想出的架构一定不会是完美的，在动工后可能会（最后也的确是）出现各种各样的问题，此时也需要自行或咨询导师进行架构的调整。
+
+![uml](/home/kxz/Documents/Study/MeMyself/Summer2020/blogs/final/images/uml.png)
```
Comment (lijiarui) [inline jekyll/_posts/2020-09-26-chassist-final.md]: Where is this image?

**Review by lijiarui — CHANGES_REQUESTED** [review]
Please change as I request:
- add tag for this blog
- add image for this blog as the blog link request.

**Comment by lijiarui** [comment]
@wj-Mcat  Please help @kxz18  posting his blog.

**Comment by wj-Mcat** [comment]
> Please change as I request:
> 
> * add tag for this blog
> * add image for this blog as the blog link request.

Please add these two main features for this blog, thanks a lot. Waiting for your new commit.

**Comment by kxz18** [comment]
> > Please change as I request:
> > 
> > * add tag for this blog
> > * add image for this blog as the blog link request.
> 
> Please add these two main features for this blog, thanks a lot. Waiting for your new commit.

Both have been done in commits of yesterday.

**Review by lijiarui — APPROVED** [review]
Good job!

**Pull request merged**